whylog/config/__init__.py

```diff
@@ -13,6 +17,89 @@
 from whylog.config.rule import RegexRuleFactory, Rule
 
 
+class ConfigFactory(object):
```

Comment:
This is `AbstractConfigFactory` and `YamlConfigFactory` mixed together. Split it.